design/FY2021/NFP-Spaces.md

```diff
@@ -0,0 +1,234 @@
+    * Revised to reflect simplified Construction:AirBoundary object (radiant and solar enclosures are now the same)
+
+## Justification for New Feature ##
+
+Thermal zones (HVAC zones) are often composed of a variety of spaces grouped together to be served by
+a single thermostat. For example, an office building core zone may include private offices, conference rooms, restrooms
+and corridors. Each of these spaces (rooms) has different internal gains and may or may not be fully enclosed by solid surfaces.
+Currently, the smallest building element is a zone. This requires the user (or interface) to blend these spaces together for surfaces,
+internal gains, and other specifications. The option to specify each space explicitly would greatly simplify input in data managemenent,
+especially for space-based interfaces (such as OpenStudio) and for codes and standards modeling.
+
+There are also computing performance advantages to dividing zones into smaller enclosures for radiant exchange.
+
+## Overview ##
+
+### Current EnergyPlus model ###
+EnergyPlus input currently has the following (very flat) heierarchy:
+
+    Zone
+        Surfaces (references a zone name)
+        People, Lights, etc. (references a zone name or zone list name)
+        Thermostat (one per zone)
+        HVAC equipment (attaches to a zone)
+
+Zones represent an air mass (air node) that exchanges heat with surfaces, internal loads, HVAC equipment, etc. 
+A Zone is essentially the "atomic unit" of the building. By default a zone has a uniform air temperature, but there are room air models 
+that allow modeling of stratification and other non-uniform temperature effects. Thermostats and HVAC equipment are assigned per Zone.
+
+The internal data model adds another layer:
+
+    Enclosure
+        Zone
+            Surfaces
+            People, Lights, etc.
+            Thermostat
+            HVAC equipment
+        Zone
+            Surfaces
+            People, Lights, etc.
+            Thermostat
+            HVAC equipment
+
+Enclosures are used for radiant and solar/daylighting exchange. Enclosures are primarily concerned with Surfaces, but they
+are also related to internal gains (which cast radiant and visible energy into the enclosure). By default there is one Enclosure for each
+Zone.  By using Construction:AirBoundary, multiple zones may be grouped into a single larger Enclosure. 
+Enclosures are assigned automatically based on the zones connected by an air boundary surface. 
+
+
+## New Space Object
+
+### Relationship assumptions ##
+
+If we keep the current definition of "Zone" and add the concept of
+"Space" (<= Zone), then this would add a new layer to the data model:
+
+    Zone
+        Thermostat
+        HVAC equipment
+        Space 2
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+        Space 3
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+
+    Enclosure
```

Comment:
This doesn't look like a correct tree hierarchy to me.  Presumably some of the surfaces under an Enclosure are shared by a Zone?  Should Enclosure be underneath Zone or vice versa?

Reply:
I think the proposed hierarchy is conceptual and conceptually matches OS.

Reply:
![image](https://user-images.githubusercontent.com/1327850/100272625-b823b300-2f18-11eb-8317-5bbd617b11ed.png)